ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import pluralize from 'pluralize';
+
+import { exceptionManagementPath } from 'routePaths';
+import {
+    VulnerabilityDeferralException,
+    VulnerabilityException,
+    VulnerabilityFalsePositiveException,
+} from 'services/VulnerabilityExceptionService';
+import { getDate, getDistanceStrictAsPhrase } from 'utils/dateUtils';
+
+// @TODO: Add tests for these
+
+function isVulnerabilityDeferralException(
+    exception: VulnerabilityException
+): exception is VulnerabilityDeferralException {
+    return exception.targetState === 'DEFERRED';
+}
+
+function isVulnerabilityFalsePositiveException(
+    exception: VulnerabilityException
+): exception is VulnerabilityFalsePositiveException {
+    return exception.targetState === 'FALSE_POSITIVE';
+}
+
+export type RequestIDTableCellProps = {
+    id: VulnerabilityException['id'];
+    name: VulnerabilityException['name'];
+};
+
+export function RequestIDTableCell({ id, name }: RequestIDTableCellProps) {
+    return <Link to={`${exceptionManagementPath}/requests/:${id}`}>{name}</Link>;
+}
+
+export type RequesterTableCellProps = {
+    requester: VulnerabilityException['requester'];
+};
+export function RequesterTableCell({ requester }: RequesterTableCellProps) {
+    return <div>{requester.name}</div>;
+}
+
+export type RequestedActionTableCellProps = {
+    exception: VulnerabilityException;
+};
+
+function getRequestedAction(exception: VulnerabilityException): string {
+    if (isVulnerabilityDeferralException(exception)) {
+        const latestExpiry = exception.deferralUpdate
+            ? exception.deferralUpdate.expiry
+            : exception.deferralReq.expiry;
```

Comment:
I think we want to handle this differently depending on what tab the user is viewing.

IMO:

- If there is no deferral update, simply show `Deferred (<original duration>)`
- If there is an unapproved deferral update:
-- Show the requested action as `Deferral pending update (<update duration>)` when on the Pending tab
-- Show the requested action as `Deferred (<original duration>)` when on the approved tab (this deviates from what we currently show in Risk Acceptance)

The thought here is that when viewing the Approved tab, the table will show what action the request was approved for, while still showing what action is pending on the Pending tab.

@mansursyed What do you think about the text in the Approved/Pending tabs when a request was approved, but has an unapproved update?

Reply:
If you approve a deferral, and then update it, I thought it would go back to the `Pending approvals` tab. The second sub-bullet in the second bullet makes it seem like it would stay in the `Approved deferrals` tab.

Reply:
Oh actually, I'm remembering things again. So we actually show the updated deferral in both tabs, don't we? That's probably what you meant

Reply:
Okay so these are the states based on what you mentioned above:

- Pending approvals tab
    - `targetState = DEFERRAL & exceptionStatus = PENDING` -> Use `deferralReq`
    - `targetState = DEFERRAL & exceptionStatus = APPROVED_PENDING_UPDATE` -> Use `deferralUpdate`
- Approved Deferrals
    - `targetState = DEFERRAL & exceptionStatus = APPROVED` -> Use `deferralReq`
    - `targetState = DEFERRAL & exceptionStatus = APPROVED_PENDING_UPDATE` -> Use `deferralReq`

Reply:
We also now have a `Denied requests` tab. So I'm wondering what we should be showing for that one. If we defer a CVE for 30 days, approve it, update it to 90 days, and then deny it, what happens?